packages/server/src/fhir/operations/utils/terminology.ts

```diff
@@ -31,23 +32,36 @@ export async function findTerminologyResource<T extends TerminologyResource>(
     ],
   });
 
-  if (!resources.length) {
+  if (!results.length) {
     throw new OperationOutcomeError(badRequest(`${resourceType} ${url} not found`));
-  } else if (resources.length === 1) {
-    return resources[0];
+  } else if (results.length === 1) {
+    return results[0];
   } else {
-    resources.sort((a: TerminologyResource, b: TerminologyResource) => {
-      // Select the non-base FHIR versions of resources before the base FHIR ones
-      // This is kind of a kludge, but is required to break ties because some CodeSystems (including SNOMED)
-      // don't have a version and the base spec version doesn't include a date (and so is always considered current)
-      if (a.meta?.project === r4ProjectId) {
-        return 1;
-      } else if (b.meta?.project === r4ProjectId) {
-        return -1;
+    const resourceReferences: Reference<T>[] = [];
```

Comment:
Rather than building array and sorting, you can just linearly walk the project list.

```suggestion
    const projectResource = results.find((r) => r.meta?.project === project.id);
    if (projectResource) {
      return projectResource;
    }
    if (project.link) {
      for (const linkedProject of project.link) {
        const linkedResource = results.find((r) => r.meta?.project === resolveId(linkedProject.project));
        if (linkedResource) {
          return linkedResource;
        }
      }
    }
    throw new OperationOutcomeError(badRequest(`${resourceType} ${url} not found`));
```